pkg/controller/node/node_controller.go

```diff
@@ -554,22 +554,25 @@ func getCandidateMachines(pool *mcfgv1.MachineConfigPool, nodesInPool []*corev1.
 }
 
 func maxUnavailable(pool *mcfgv1.MachineConfigPool, nodes []*corev1.Node) (int, error) {
-	// Hardcoded for now per discussion in
-	// https://github.com/openshift/machine-config-operator/pull/701
-	if pool.Name == "master" {
-		return 1, nil
-	}
 	intOrPercent := intstrutil.FromInt(1)
 	if pool.Spec.MaxUnavailable != nil {
 		intOrPercent = *pool.Spec.MaxUnavailable
 	}
-
 	maxunavail, err := intstrutil.GetValueFromIntOrPercent(&intOrPercent, len(nodes), false)
 	if err != nil {
 		return 0, err
 	}
 	if maxunavail == 0 {
 		maxunavail = 1
 	}
+	if pool.Name == "master" {
+		// calculate the fault tolerance dynamically for the master pool
+		// to avoid risking losing etcd quorum.
+		tolerance := len(nodes) - ((len(nodes) / 2) + 1)
```

Comment:
While we're here, let's also warn if the number of nodes isn't odd? 
https://github.com/etcd-io/etcd/blob/master/Documentation/faq.md#what-is-failure-tolerance

Reply:
:thinking: should we warn here? isn't that something well documented in the etcd docs? having an even number doesn't mean you're doing something wrong per-se, just that you need a wider quorum for less total nodes and no real gain

Reply:
just to share more on my take: I believe that if someone is looking for a warning in the MCC logs, they already have a cluster up and running which might be not perfectly configured with an odd number of masters/etcd. But the warning, in general, is a good point, maybe it should be something in the installer so that people trying to use 4 masters are warned that the size isn't _optimal_.